File: examples/pizzas_domain/lib/pizzas_domain/order/events/placed_order.rb
module PizzasDomain
  class Order
    module Events
      class PlacedOrder
        attr_reader :aggregate_id, :customer_name, :pizza, :quantity, :items, :status, :occurred_at

        def initialize(aggregate_id: nil, customer_name: nil, pizza: nil, quantity: nil, items: nil, status: nil)
          @aggregate_id = aggregate_id
          @customer_name = customer_name
          @pizza = pizza
          @quantity = quantity
          @items = items
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: pizzas_static_go/domain/placed_order.go
package domain

import "time"

type PlacedOrder struct {
	AggregateID string `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	CustomerName string `json:"customer_name"`
	Quantity int64 `json:"quantity"`
	Items []OrderItem `json:"items"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (p PlacedOrder) EventName() string { return "PlacedOrder" }

func (p PlacedOrder) GetOccurredAt() time.Time { return e.OccurredAt }
